rules/terraformrules/terraform_module_version.go

```diff
@@ -0,0 +1,118 @@
+package terraformrules
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/terraform-linters/tflint/terraform/addrs"
+	"github.com/terraform-linters/tflint/terraform/configs"
+	"github.com/terraform-linters/tflint/tflint"
+)
+
+var exactVersionRegexp = regexp.MustCompile(`^=?\s*\d+\.\d+\.\d+$`)
```

Comment:
I'm a little worried if this regex is enough. For instance, go-version uses the following regexp:
https://github.com/hashicorp/go-version/blob/v1.3.0/version.go#L18-L31

Ideally, it would be nice to be able to verify that the `f` in the `Constraint` is the `constraintEqual`, but this struct doesn't seem to expose the logic...
https://github.com/hashicorp/go-version/blob/v1.3.0/constraint.go#L30-L31

Reply:
Agree. Spent a bunch of time looking at constraints and saw that the constraint type was not exposed. Was thinking of this more complicated regexp compilation:

https://github.com/hashicorp/go-version/blob/c2de2312d363381e4254abec260c6df2b1226483/constraint.go#L45-L49

But the version one is easy to reuse.

Reply:
After trying this, the go-version regex doesn't require major/minor/patch and the test ensuring `1.0` is rejected as exact fails. Grabbed a regex from the semver.org docs and added a test ensuring that prerelease suffixes work.